### ISSUE TITLE
"InputFormatException: Invalid data found when processing input" Error when processing uncorrupted file

### ISSUE THREAD

**Issue by SpazmasterYT** [a-schild/jave2#96]
I've received this error 95% of the time when processing video to wav audio conversion for any video file. I am using the latest version of Jave2 and amd64 ffmpeg that comes with jave2. Sometimes it will seem I am lucky and it will successfully convert a video into wav file, but most of the times it will fail with this error. My videos are not corrupt, they are simple mp4 files that play on any video player. Sometimes I will get lucky and a video that I test it on will fail a few times then randomly work one time then continue to fail. 

My code that involves conversion is as follows:

```
public class MediaConverter {
    private Encoder encoder = new Encoder();
    private AudioAttributes audio;
    private EncodingAttributes attrs;
    private char fs = File.separatorChar;

    
    
     public MediaConverter(){

        audio = new AudioAttributes();
        
        audio.setCodec("pcm_s16le");
        attrs = new EncodingAttributes();
        attrs.setFormat("wav");
        attrs.setAudioAttributes(audio);
        
        
    }
     
     
     private void convert(File input) throws IllegalArgumentException, EncoderException{
         File target = new File("tmp"+fs+"audio.wav");
         
         encoder.encode(new MultimediaObject(input), target, attrs);
         
         
     }
     
     
     public boolean test(File input){
         
        try {
            convert(input);
        } catch (IllegalArgumentException ex) {
            Logger.getLogger(MediaConverter.class.getName()).log(Level.SEVERE, null, ex);
        } catch (EncoderException ex) {
            Logger.getLogger(MediaConverter.class.getName()).log(Level.SEVERE, null, ex);
        }
         
         return false;
     }

    
}
```
The error message catches at encoder.encode(new MultimediaObject(input), target, attrs);

**Comment by a-schild** [comment]
* Can you reproduce it with a specific media file (Which you can share with us) 
* Do you have the full details of the error message which is thrown?
* Can you please also turn on Debug/Trace Loggin on the ws.schild.jave.MultimediaObject and tell me what messages are received

**Comment by SpazmasterYT** [comment]
I found the likely cause to why this error happened, turns out the API I was using that downloaded files that would be passed to this encode was non-blocking (so stupid of me to not realize this, such a simple reason that I completely overlooked when trying to debug this). Basically, when the downloading method is called from this other API I use, rather than block until the file is done downloading(which would make logical sense but it doesn't do that for some reason), it simply passes to the next instruction, meaning by the time the ffmpeg object receives the file, the file is not done downloading so it throws this invalid data found error because ffmpeg thinks its corrupt. I found this out by using Thread.sleep(3000) in between downloading and encoding.

For those in the future that may have this problem, check to make sure whatever way you are downloading has enough time to write the file and close it before attempting to open it with the ffmpeg encoder.
